app/controllers/registrations_controller.rb

```diff
@@ -147,11 +147,20 @@ def add
 
   def do_add
     @competition = competition_from_params
+
     if @competition.registration_full?
       flash[:danger] = I18n.t("registrations.mailer.deleted.causes.registrations_full")
       redirect_to competition_path(@competition)
       return
```

Comment:
(sidenote, as it's not strictly part of the PR but yay consistency!)
```suggestion
      return redirect_to competition_path(@competition)
```